tests/ignite/metrics/test_ssim.py

```diff
@@ -0,0 +1,148 @@
+import os
+
+import numpy as np
+import pytest
+import torch
+from skimage.measure import compare_ssim as ski_ssim
+
+import ignite.distributed as idist
+from ignite.exceptions import NotComputableError
+from ignite.metrics import SSIM
+
+
+def test_zero_div():
+    ssim = SSIM(data_range=1.0)
+    with pytest.raises(NotComputableError):
+        ssim.compute()
+
+
+def test_invalid_ssim():
+    y_pred = torch.rand(16, 1, 32, 32)
+    y = y_pred + 0.125
+    with pytest.raises(ValueError):
+        ssim = SSIM(data_range=1.0, kernel_size=10)
+        ssim.update((y_pred, y))
+        ssim.compute()
+
+    with pytest.raises(ValueError):
+        ssim = SSIM(data_range=1.0, kernel_size=-1)
+        ssim.update((y_pred, y))
+        ssim.compute()
+
+    with pytest.raises(ValueError):
+        ssim = SSIM(data_range=1.0, kernel_size=1.0)
+        ssim.update((y_pred, y))
+        ssim.compute()
+
+    with pytest.raises(ValueError):
+        ssim = SSIM(data_range=1.0, sigma=-1)
+        ssim.update((y_pred, y))
+        ssim.compute()
+
+    with pytest.raises(ValueError):
+        ssim = SSIM(data_range=1.0, sigma=1)
+        ssim.update((y_pred, y))
+        ssim.compute()
+
+
+def test_ssim():
+    ssim = SSIM(data_range=1.0)
+    device = "cuda" if torch.cuda.is_available() else "cpu"
+    y_pred = torch.rand(16, 3, 32, 32, device=device)
+    y = y_pred + 0.125
+    ssim.update((y_pred, y))
+
+    np_pred = np.random.rand(16, 32, 32, 3)
+    np_y = np.add(np_pred, 0.125)
+    np_ssim = ski_ssim(np_pred, np_y, win_size=11, multichannel=True, gaussian_weights=True)
+
+    assert isinstance(ssim.compute(), torch.Tensor)
+    assert torch.allclose(
+        ssim.compute(), torch.tensor(np_ssim, dtype=torch.float32, device=device), atol=1e-3, rtol=1e-3
+    )
+
+
+def _test_distrib_integration(device, tol=1e-3):
+    from ignite.engine import Engine
+
+    rank = idist.get_rank()
+    n_iters = 100
+    s = 10
+    offset = n_iters * s
+
+    y_pred = torch.rand(offset * idist.get_world_size(), 3, 28, 28, dtype=torch.float, device=device)
+    y = y_pred + 0.125
+
+    def update(engine, i):
+        return (
+            y_pred[i * s + offset * rank : (i + 1) * s + offset * rank],
+            y[i * s + offset * rank : (i + 1) * s + offset * rank],
+        )
+
+    engine = Engine(update)
+    SSIM(data_range=1.0).attach(engine, "ssim")
+
+    data = list(range(n_iters))
+    engine.run(data=data, max_epochs=1)
+
+    assert "ssim" in engine.state.metrics
+    res = engine.state.metrics["ssim"]
+
+    np_pred = np.random.rand(offset * idist.get_world_size(), 28, 28, 3)
```

Comment:
Shouldn't we use `y_pred` and `y` to setup `np_pred` and `np_true` ?

Reply:
Since `skimage` takes the last index as channels, I re-created it with numpy

Reply:
Technically, it is not the same data. Otherwise, you can transpose the tensor.

Reply:
Yes, it wouldn't be the same data.

But the difference between `y_pred` and `y` and that of `np_pred` and `np_true` is the same.
So, the computed ssim would be similar.

Reply:
Yes, I understood that from the code. Let's compare the same data.

Reply:
Okay
should I change to the same data for all tests? or only the distributed tests?

Reply:
Yes, it would be better for all the tests

Reply:
Okay, will change that.